test/test_torch.py

```diff
@@ -18056,6 +18077,73 @@ def test_count_nonzero(self, device, dtype):
         self._test_reduction_function_with_numpy(torch.count_nonzero, np.count_nonzero, device, dtype)
         self._test_reduction_function_with_numpy(torch.count_nonzero, np.count_nonzero, device, dtype, True)
 
+    def _test_sum_reduction_vs_numpy(self, torch_fn, np_fn, device, dtype, with_keepdim=False, with_extremal=False):
+        def is_integral(dtype):
+            return dtype in torch.testing.get_all_int_dtypes()
+
+        # On Windows CI, the current version of `numpy` promotes all lower integers
+        # dtypes to int32 while `torch` promotes them to int64. Hence we skip on checking
+        # the exact dtype.
+        # Reference : https://dr.pytorch.org/api/view-log-full?build_id=122051580
+        # PR : https://github.com/pytorch/pytorch/pull/38628#issuecomment-655905370
+        exact_dtype = False if (IS_WINDOWS and is_integral(dtype)) else True
+
+        if dtype == torch.uint8:
+            with self.assertRaises(TypeError):
+                self._test_reduction_function_with_numpy(torch_fn, np_fn, device, dtype, with_extremal=with_extremal)
+        else:
+            # TODO: Investigate why the output is not close to numpy.
+            if dtype == torch.float16:
+                self._test_reduction_function_with_numpy(torch_fn, np_fn, device, dtype,
+                                                         atol=0.4, rtol=1e-2, exact_dtype=exact_dtype,
+                                                         with_keepdim=with_keepdim, with_extremal=with_extremal)
+            elif dtype == torch.float32:
+                self._test_reduction_function_with_numpy(torch_fn, np_fn, device, dtype,
+                                                         atol=7e-05, rtol=3e-06, exact_dtype=exact_dtype,
+                                                         with_keepdim=with_keepdim, with_extremal=with_extremal)
+            else:
+                self._test_reduction_function_with_numpy(torch_fn, np_fn, device, dtype,
+                                                         exact_dtype=exact_dtype, with_keepdim=with_keepdim)
+
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(*(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes(include_bfloat16=False)))
+    def test_sum_vs_numpy(self, device, dtype):
+        self._test_sum_reduction_vs_numpy(torch.sum, np.sum, device, dtype)
+        self._test_sum_reduction_vs_numpy(torch.sum, np.sum, device, dtype, with_extremal=True)
+        # Reference: https://dr.pytorch.org/api/view-log-full?build_id=122265220
+        # RuntimeError: Check failed: status.status() == ::tensorflow::Status::OK()
+        # (Invalid argument: Reshape operation has mismatched element counts: from=1 (s64[]) to=10 (s64[10,1]) vs. OK)
+        if self.device_type != 'xla':
```

Comment:
You can use the @onlyOnCPUAndCUDA decorator instead of this check, which will prevent this test from running on XLA